Source/Charts/Data/Implementations/Standard/ChartData.swift

```diff
@@ -757,3 +764,138 @@ open class ChartData: NSObject
         return max
     }
 }
+
+// MARK: MutableCollection
+extension ChartData: MutableCollection
+{
+    public typealias Index = Int
+    public typealias Element = ChartDataSetProtocol
+
+    public var startIndex: Index
+    {
+        return _dataSets.startIndex
+    }
+
+    public var endIndex: Index
+    {
+        return _dataSets.endIndex
+    }
+
+    public func index(after: Index) -> Index
+    {
+        return _dataSets.index(after: after)
+    }
+
+    public subscript(position: Index) -> Element
+    {
+        get{ return _dataSets[position] }
+        set{ self._dataSets[position] = newValue }
+    }
+}
+
+// MARK: RandomAccessCollection
+extension ChartData: RandomAccessCollection
+{
+    public func index(before: Index) -> Index
```

Comment:
I saw `RandomAccessCollection` has more required methods to provide, but here only one `index(before: Index)` which just comes from `BidirectionalCollection`. What knowledge I missed here?

Reply:
From the `RandomAccessCollection` doc:
> The RandomAccessCollection protocol adds further constraints on the associated Indices and SubSequence types, but otherwise imposes no additional requirements over the BidirectionalCollection protocol. However, in order to meet the complexity guarantees of a random-access collection, either the index for your custom type must conform to the Strideable protocol or you must implement the index(_:offsetBy:) and distance(from:to:) methods with O(1) efficiency